t/t3907-stash-show-config.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/sh
+
+test_description='Test git stash show configuration.
+
+'
+
+. ./test-lib.sh
+
+config_stash_stat_and_patch () {
+	if test "$1" != "unset"
+	then
+		git config stash.showStat "$1"
+	fi
+
+	if test "$2" != "unset"
+	then
+		git config stash.showPatch "$2"
+	fi
+}
+
+unset_config () {
+	if git config --bool stash.showStat >/dev/null
+	then
+		git config --unset stash.showStat
+	fi
+
+	if git config --bool stash.showPatch >/dev/null
+	then
+		git config --unset stash.showPatch
+	fi
+}
+
+get_expected_output () {
+	FLAGS=
+
+	if test "$(git config --bool stash.showStat || echo true)" = "true"
+	then
+		FLAGS=--stat
+	fi
+
+	if test "$(git config --bool stash.showPatch || echo false)" = "true"
+	then
+		FLAGS=${FLAGS}${FLAGS:+ }-p
+	fi
+
+	git diff $FLAGS >expect
+}
+
+get_actual_output () {
+	git stash &&
+	git stash show >actual
+}
+
+check_output () {
+	get_expected_output &&
+	get_actual_output &&
+	test_cmp expect actual &&
+	rm expect actual
+}
+
+options="unset true false"
+
+test_expect_success 'setup' '
+	echo 1 >file &&
+	git add file &&
+	git commit -m initial
```

Comment:
Shorter: `test_commit file` (the file name will be `file.t` in that case, though).